optimum/habana/transformers/models/clip/modeling_clip.py

```diff
@@ -0,0 +1,19 @@
+import torch
+from transformers.models.clip.modeling_clip import CLIPVisionEmbeddings
+
+
+class GaudiCLIPVisionEmbeddings(CLIPVisionEmbeddings):
+    def forward(self, pixel_values: torch.FloatTensor) -> torch.Tensor:
+        batch_size = pixel_values.shape[0]
+        target_dtype = self.patch_embedding.weight.dtype
+        if "float8" in str(target_dtype):
```

Comment:
add a condition to check  self.patch_embedding.weight/pixel_values are on hpu

Reply:
Done

Reply:
i meant 
if "float8" in str(target_dtype) and pixel_values.device.type == "hpu":

Reply:
Updated